### PR TITLE
Release amendments and GitHub Release changelog improvements

### PR DESCRIPTION
**Pull request by ameteiko**
## Description

This pull request amends the release process by addressing the bug in the changelog containing features from previous releases.

Changes proposed in this pull request:
- Amendments to the release process:
  - The main branch incorporates commits that bump the dependent containers.
  - The release tags point to the commit in the release branches and the corresponding commits in the main branch.
- GitHub Release's changelog features grouping:
  - `Test suites` section aggregates all changes containing a `test` word.
  - `Documentation` section aggregates all changes containing `docs` or `document` word fragments. 
  - `Features` section contains the rest of the changes.

## Related Issues and Documents

Closes:
#287 

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [ ] New features have a milestone label set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:
- [x] No, because unit tests are not needed.

The feature is e2e-tested:
- [x] No, because e2e-tests are not needed.

Tests conducted for the PR:
Release process on an imported repo

## Codebase
Irrelevant

### CODE REVIEW

File: docs/contributor/releasing.md [hunk 1]
```diff
@@ -1,41 +1,40 @@
...
+   Create a PR to `telemetry-manager/main` with the following changes:
+   - `Makefile`:
+      - Ensure the `IMG` variable reflects the latest `telemetry-manager` version.
+      - Update the `MODULE_VERSION` variable to the value of the new module version.
+   - `config/manager/kustomization.yaml`:
+      - Ensure the `newTag` field for the `telemetry-manager` image reflects the latest version.
+   - `sec-scanners-config.yaml`:
+      - Ensure that all images listed in the `protecode` field have the same versions as those used in the `main.go` file.
+
+2. create a release branch in the `telemetry-manager` repository after merging the PR.
```
Comment (NHingerl) [inline docs/contributor/releasing.md:15]:
```suggestion
2. Merge the PR.
3. In the `telemetry-manager` repository, create a release branch.
```

File: docs/contributor/releasing.md [hunk 2]
```diff
@@ -1,41 +1,40 @@
 # Release Process
 
-1. For new major and minor versions: Create a release branch in the `telemetry-manager` repository. The name of this branch must follow the `release-x.y` pattern, such as `release-1.0`.
-
-
-   ```bash
+This release process covers the steps to release new major and minor versions for the Kyma Telemetry Component.
+
+1. Bump the `telemetry-manager/main` branch with the new versions for the dependent images.
+   Create a PR to `telemetry-manager/main` with the following changes:
```
Comment (NHingerl) [inline docs/contributor/releasing.md:6]: Considering you specified the repo in the following branch, why not here as well?

File: docs/contributor/releasing.md [hunk 3]
```diff
@@ -1,41 +1,40 @@
...
-2. Create a PR to `telemetry-manager/release-x.y` with the following changes:
-    - `Makefile`:
-        - Ensure that the `IMG` variable is up to date with the latest `telemetry-manager` version.
-        - Update the `MODULE_VERSION` variable to the value of the new module version.
-    - `config/manager/kustomization.yaml`:
-        - Ensure that the `newTag` field for the `telemetry-manager` image is up to date with the latest version.
-    - `sec-scanners-config.yaml`:
-        - Ensure that all of the images listed in the `protecode` field have the same versions as the images used in the `main.go` file.
-3. After merging the PR, create a tag on the release branch that has the value of the new module release version.
+3. Create a release tag for the head commit in the `telemetry-manager/{RELEASE_BRANCH}` branch.
```
Comment (NHingerl) [inline docs/contributor/releasing.md:23]:
```suggestion
3. In the `telemetry-manager/{RELEASE_BRANCH}` branch, create a release tag for the head commit.
```

File: docs/contributor/releasing.md [hunk 4]
```diff
@@ -1,41 +1,40 @@
...
-4. To trigger a postsubmit job (`post-telemetry-manager-release-module`) that creates the GitHub release, push the tag.
-
-    ```bash
-    git push upstream {RELEASE_VERSION}
-    ```
+   git push upstream {RELEASE_VERSION}
+   ```
 
 5. Verify the [Prow Status](https://status.build.kyma-project.io/) of the postsubmit job (`post-telemetry-manager-release-module`).
    - Once the postsubmit job succeeds, the new Github release should be available under [releases](https://github.com/kyma-project/telemetry-manager/releases).
```
Comment (NHingerl) [inline docs/contributor/releasing.md:35]: Why "should"? Where else could it end up after the postsubmit job succeeded? 

```suggestion
   - Once the postsubmit job succeeds, the new Github release is available under [releases](https://github.com/kyma-project/telemetry-manager/releases).
```

File: docs/contributor/releasing.md [hunk 5]
```diff
@@ -1,41 +1,40 @@
...
    - If the postsubmit job failed, you can re-trigger it by removing the tag from upstream and pushing it again:
```
Comment (NHingerl) [inline docs/contributor/releasing.md:36]: Why "can"? Is retriggering it optional? What other options are there?
If retriggering is the only way to proceed, I suggest:

```suggestion
   - If the postsubmit job failed, retrigger it by removing the tag from upstream and pushing it again:
```

File: docs/contributor/releasing.md (hunk 2 above)
Comment (NHingerl) [reply docs/contributor/releasing.md:6]: disregard this; of course it's written here.

**Review by NHingerl — APPROVED** [review]

**Review by shorim — APPROVED** [review]

**Pull request merged**